### ISSUE TITLE
Test #65 (and regression)

### ISSUE THREAD

**Issue by patrickmj** [omeka-s-modules/CSVImport#68]
Big thing here is #65. It also involved moving a lot around, and there have been other changes lately, so some regression checking that old CSV import files are still happy might be in order. Especially important is the combination of different media imports (html, youtube, oembed, etc), and multivalue imports for the new resource relations option.

Use develop branch of omeka-s, and branch resource-reference of CSVImport.

The Resource Reference option actually goes beyond the original request, including all Resources, not just Items. Thus, the IDs could refer to Items, Item Sets, or Media (or other Resources if something adds them, which I don't think we do yet). Thus, the possible interpretations of property values is a little wider open than the admin interface, which only has options for Item and Item Set.

The "Import as Omeka S Resource ID" option is new under the property selector.

**Comment by mebrett** [comment]
How should the IDs be formatted? Just the number or item/[id] and item-set/[id]?

**Comment by patrickmj** [comment]
Just the number -- doesn't matter whether its an item or an item set because the ids are both in the same Resource table.

**Comment by mebrett** [comment]
In order for it to work to import the reference, you have to map the column in the CSV both to a property and then map it as an ID (see below). Is that expected behavior?


![screen shot 2017-03-13 at 3 59 30 pm](https://cloud.githubusercontent.com/assets/2314070/23872835/2ba6a460-0806-11e7-879a-8fc6b1f91f66.png)

**Comment by patrickmj** [comment]
Yep. It the same with URL references and the regular item add page

**Comment by mebrett** [comment]
Importer on latest pull is throwing error (even on CSVs which have worked in the past): 
```
Fatal error: Argument 4 passed to Omeka\Api\Manager::batchCreate() must be of the type array, boolean given, called in /websites/omekadev/home/www/omeka-s/modules/CSVImport/src/Job/Import.php on line 85 and defined
in /websites/omekadev/home/www/omeka-s/application/src/Api/Manager.php on line 93
```

**Comment by patrickmj** [comment]
I can't reproduce this on the latest develop branch.

Could you include the entire stack trace?

And, please tell me more details to reproduce it, including sending me the CSV file you are using and the mapping settings.

What do you see that makes it look like a change to the develop branch?

**Comment by patrickmj** [comment]
Nevermind. reproduced this (latest develop of S). Much fixing to happen before we can come back to this.

**Comment by patrickmj** [comment]
Latest develop branch of CSVImport should have it working with the API changes in core Omeka S.

**Comment by mebrett** [comment]
How would html media be imported in a csv? (trying to figure out how to test that)

**Comment by patrickmj** [comment]
Should just be putting HTML into the column, and choosing HTML from the Media Import stuff on the mapping page

**Comment by mebrett** [comment]
Thanks - I was overthinking it.

**Comment by mebrett** [comment]
A bad media entry (invalid oEmbed or IIIF, for example) will prevent the entire row from being imported. [example log](http://dev.omeka.org/omeka-s/admin/job/413/log)

**Comment by patrickmj** [comment]
Can't currently get to the site to look at the log, but that seems to me like as it should be -- if something goes wrong, the item shouldn't be imported so people can try to fix it and get a clean import? Otherwise, it'll look like everything is fine and they'll only notice it later.

**Comment by patrickmj** [comment]
Got to the site (guess there's iffy internet today at home!). That log entry looks as it should be to me, and looks like the fix in import 414 was good.

Also worth noting that that site is on Omeka S beta2, but lots of API changes are in beta3, so updating the site will be a good idea.

**Comment by mebrett** [comment]
Ah, the site was still on the import-template branch from testing changes to the core yesterday.

**Comment by patrickmj** [comment]
so many branches, so little time! I've also fallen into that complication.

**Comment by mebrett** [comment]
On latest branch of develop for core and module - undos are getting stuck at "undo in progress"
http://dev.omeka.org/omeka-s/admin/csvimport/past-imports

**Comment by patrickmj** [comment]
latest develop branch of CSVImport should fix the undo

**Comment by mebrett** [comment]
Issues with undos getting stuck is resolved.

**Comment by patrickmj** [comment]
How does it work with multiple values?

**Comment by mebrett** [comment]
Works fine with multiple values.

**Comment by mebrett** [comment]
Is there anything further to test on this?

**Comment by patrickmj** [comment]
Not that I can think of. Anything new is worth being a new issue.
